### PR TITLE
Close and let GC QueryRunners in tests

### PR DESCRIPTION
**Pull request by findepi**
Related to https://github.com/prestodb/presto/issues/10812

For most tests this is done automatically in `AbstractTestQueryFramework` (https://github.com/prestodb/presto/blob/master/presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueryFramework.java#L85-L89). For others, we need to take care manually.

### CODE REVIEW

File: presto-benchmark/src/test/java/com/facebook/presto/benchmark/BenchmarkDecimalAggregation.java [hunk 1]
```diff
@@ -66,10 +67,18 @@ public final MemoryLocalQueryRunner getQueryRunner()
...
         public void setUp()
         {
+            queryRunner = new MemoryLocalQueryRunner();
             queryRunner.execute(format(
                     "CREATE TABLE memory.default.orders AS SELECT orderstatus, cast(totalprice as %s) totalprice FROM tpch.sf1.orders",
                     type));
         }
+
+        @TearDown
+        public void tearDown()
```
Comment (sopel39) [inline presto-benchmark/src/test/java/com/facebook/presto/benchmark/BenchmarkDecimalAggregation.java]: This commit seems unneeded as benchmarks are one-time run only and are not part of Travis automation

Comment (findepi) [reply presto-benchmark/src/test/java/com/facebook/presto/benchmark/BenchmarkDecimalAggregation.java]: Some benchmarks do tear down. I don't see why this one shouldn't.

File: presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java [hunk 2]
```diff
@@ -154,7 +154,7 @@
     private FunctionAssertions functionAssertions;
     private List<ListenableFuture<?>> futures;
 
-    @BeforeSuite
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java]: It looks like the commit message is about something different.

Comment (findepi) [reply presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java]: No it's not. `@AfterSuite` happens too late (when all tests finish), so it's as there was no tear down method at all.

**Pull request merged**